src/consumer/admin.rs

```diff
@@ -0,0 +1,56 @@
+use anyhow::Error;
+use rdkafka::{
+    admin::{AdminClient, AdminOptions, NewTopic, TopicReplication},
+    metadata::Metadata,
+    ClientConfig,
+};
+use std::time::Duration;
+use tracing::info;
+
+pub async fn create_missing_topics(
+    kafka_client_config: ClientConfig,
+    topics: &[&str],
+    default_topic_partitions: i32,
+) -> Result<(), Error> {
+    let admin_client: AdminClient<_> = kafka_client_config.create().unwrap();
+    let mut topics_to_create = Vec::new();
+    for topic in topics {
+        // Weirdly, this call will actually create the topic if it doesn't exist. So it needs to be overridden in the next steps.
```

Comment:
It is very weird. What I experienced was: without any of this code, the consumer crashloops with the topic not existing. If all this function did was call `fetch_metadata`, then the crashloops would be solved.

The issue is the partitions and settings it uses in that case. I thought it would be better to explicitly create the topic with the settings we want, rather than assume whatever `fetch_metadata` does will work properly.

Reply:
Yeah that makes sense, so what if you just make the call to `create_topic` instead of fetching metadata?
Like
```rust
let res = admin_client
    .create_topics(
        ...,
        &AdminOptions::new()
     )
    .await?;
match res {
    Ok() | TopicResult::Err((_, RDKafkaErrorCode::TopicAlreadyExists)) => Ok(())
    err => err
}
```